src/libraries/Common/src/Interop/Interop.ICU.cs

```diff
@@ -10,7 +10,16 @@ internal static partial class Interop
     internal static partial class Globalization
     {
         [DllImport(Libraries.GlobalizationNative, EntryPoint = "GlobalizationNative_LoadICU")]
-        internal static extern int LoadICU();
+        private static extern int LoadICUDirect();
+
+        [DllImport(Libraries.GlobalizationNative, EntryPoint = "GlobalizationNative_LoadICUData")]
+        private static extern int LoadICUData(string? path);
+
+        internal static int LoadICU()
+        {
+            object? datPath = AppContext.GetData("ICU_DAT_FILE_PATH");
```

Comment:
I was hoping you'd ask that question ;-).  Maybe.  I considered making this iOS only first, but thought I'd get your feedback and adjust if necessary.

Reply:
Is `ICU_DAT` typo or not?

I don't see why this should be mobile only

Reply:
If we want to support it on mobile only, we would need to also implement: GlobalizationNative_LoadICUData as part of this PR for non-static ICU shim.

That's why the build is failing: 
```
../Native.Unix/System.Globalization.Native/libSystem.Globalization.Native.a(entrypoints.c.o):(.data.rel.ro+0x1e8): undefined reference to `GlobalizationNative_LoadICUData'
```

I think this would only be interesting for app local ICU, because the ICU Dat file need to match the ICU version, and OSs have different versions of ICU.

cc: @tarekgh

Reply:
If we wanted to support his on non-mobile we need to implement the native entrypoint and the logic to support it. Currently we only have the logic for static icu shim (which is the one used for mobile). That's why the build is failing for non mobile with:

```
../Native.Unix/System.Globalization.Native/libSystem.Globalization.Native.a(entrypoints.c.o):(.data.rel.ro+0x1e8): undefined reference to `GlobalizationNative_LoadICUData'
```

Also, I don't think it is relevant for scenarios running on non mobile other than app local ICU because the ICU Dat file needs to match ICU's version and ICU ships with the OS. How relevant it would be for app local ICU scenario, I don't know as if you are already providing an app local ICU, the data would already be linked into ICU. It would be beneficial to support it only if people want to support different scenarios and provide different dat files like in mobile to support lighter weight app scenarios.

cc: @tarekgh

Reply:
I prefer to restrict this to the mobile platforms. We need to avoid the scenarios that the data files are not matching the code library. Also, we have the app-local feature for anyone want to use customized data.